Support memory allocator in Rust runtime

Currently we are building the Rust part of the Oak Runtime with `no_std` and no allocator. We should consider plugging in an allocator so we can at least support `liballoc`, and perhaps eventually also `std` itself.

See https://doc.rust-lang.org/alloc/ for documentation on `liballoc`.

https://github.com/apache/mesatee-sgx/tree/master/sgx_alloc is doing something similar by linking to the Intel SGX allocator, so we may be able to use that, or we should consider linking against the Asylo version if it's different. (@annasapek WDYT?)

Blocks #319

As a side note, if I compile the Rust code without `no_std` and link it against Asylo, the following symbols remain undefined:

- `__errno_location`
- `posix_spawn_file_actions_destroy`
- `posix_spawnattr_destroy`
- `__fxstat`
- etc.

cc @dingelish

<details>
<summary>full error message</summary>

```
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `<std::sys::unix::process::process_inner::<impl std::sys::unix::process::process_common::Command>::posix_spawn::PosixSpawnFileActions as core::ops::drop::Drop>::drop':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/process/process_unix.rs:317: undefined reference to `posix_spawn_file_actions_destroy'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `<std::sys::unix::process::process_inner::<impl std::sys::unix::process::process_common::Command>::posix_spawn::PosixSpawnattr as core::ops::drop::Drop>::drop':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/process/process_unix.rs:327: undefined reference to `posix_spawnattr_destroy'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::stack_overflow::imp::drop_handler':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/stack_overflow.rs:178: undefined reference to `sigaltstack'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::rand::imp::getrandom_fill_bytes':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/rand.rs:(.text._ZN3std6thread5local4fast12Key$LT$T$GT$14try_initialize17h876efb8994c07bb1E+0x23): undefined reference to `syscall'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/rand.rs:(.text._ZN3std6thread5local4fast12Key$LT$T$GT$14try_initialize17h876efb8994c07bb1E+0x2a): undefined reference to `__errno_location'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::rand::imp::fill_bytes':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/rand.rs:86: undefined reference to `__errno_location'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::thread::Thread::sleep':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/thread.rs:154: undefined reference to `__errno_location'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::os::errno':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::os::getcwd':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1/src/libcore/option.rs:(.text._ZN3std3env11current_dir17hbef0ba504c313948E+0xd5): undefined reference to `__errno_location'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o):/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: more undefined references to `__errno_location' follow
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::fs::File::truncate::{{closure}}':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/fs.rs:760: undefined reference to `ftruncate64'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::cvt':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/mod.rs:(.text._ZN3std2fs4File7set_len17hf9460068a36c8b1eE+0x30): undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/mod.rs:(.text._ZN3std2fs4File7set_len17hf9460068a36c8b1eE+0x37): undefined reference to `ftruncate64'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::fs::File::file_attr':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/fs.rs:716: undefined reference to `fstat64'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::os::errno':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::cvt':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/mod.rs:(.text._ZN3std2fs4File15set_permissions17h8955b5e48458b697E+0x25): undefined reference to `__errno_location'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::os::errno':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o):/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: more undefined references to `__errno_location' follow
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::fs::File::seek':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/fs.rs:799: undefined reference to `lseek64'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::os::errno':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::fs::File::seek':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/fs.rs:799: undefined reference to `lseek64'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::os::errno':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::fs::DirEntry::metadata':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/fs.rs:479: undefined reference to `dirfd'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/fs.rs:495: undefined reference to `fstatat64'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::os::errno':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o):/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: more undefined references to `__errno_location' follow
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::stack_overflow::imp::drop_handler':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/stack_overflow.rs:178: undefined reference to `sigaltstack'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::os::errno':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::thread::guard::get_stack_start':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/thread.rs:274: undefined reference to `pthread_getattr_np'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/thread.rs:278: undefined reference to `pthread_attr_getstack'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::stack_overflow::imp::make_handler':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/stack_overflow.rs:156: undefined reference to `sigaltstack'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/stack_overflow.rs:160: undefined reference to `sigaltstack'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::os::errno':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::net::on_resolver_failure':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/net.rs:395: undefined reference to `__res_init'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::os::errno':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::cvt':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/mod.rs:(.text._ZN3std10sys_common3net9TcpStream7connect17h1b7e9864ffa28edaE+0xb6): undefined reference to `__errno_location'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::os::errno':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o):/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: more undefined references to `__errno_location' follow
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys_common::net::UdpSocket::send_to':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys_common/net.rs:499: undefined reference to `sendto'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::os::errno':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::cvt':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/mod.rs:(.text._ZN3std10sys_common3net9UdpSocket7connect17h33931f5327761093E+0x6b): undefined reference to `__errno_location'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::os::errno':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::condvar::Condvar::init':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/condvar.rs:47: undefined reference to `pthread_condattr_setclock'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::os::errno':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o):/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: more undefined references to `__errno_location' follow
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::ext::net::UnixDatagram::send_to::inner':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/ext/net.rs:1259: undefined reference to `sendto'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::os::errno':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::fd::FileDesc::read_at::cvt_pread64':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/fd.rs:82: undefined reference to `pread64'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::os::errno':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::fd::FileDesc::write_at::cvt_pwrite64':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/fd.rs:124: undefined reference to `pwrite64'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::os::errno':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o):/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: more undefined references to `__errno_location' follow
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `<std::sys::unix::fs::ReadDir as core::iter::traits::iterator::Iterator>::next':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/fs.rs:440: undefined reference to `readdir64_r'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::os::errno':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `core::slice::<impl [T]>::as_ptr':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1/src/libcore/slice/mod.rs:(.text._ZN86_$LT$std..sys..unix..fs..ReadDir$u20$as$u20$core..iter..traits..iterator..Iterator$GT$4next17hf4401c198f4081f4E+0x118): undefined reference to `readdir64_r'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::fs::File::open_c::{{closure}}':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/fs.rs:649: undefined reference to `open64'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::os::errno':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::fs::File::open_c::{{closure}}':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/fs.rs:649: undefined reference to `open64'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::os::errno':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o):/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/mod.rs:(.text._ZN3std3sys4unix2fs8set_perm17h1a2954692e0ded88E+0xbe): more undefined references to `__errno_location' follow
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::fs::stat':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/fs.rs:1007: undefined reference to `stat64'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::os::errno':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::fs::lstat':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/fs.rs:1028: undefined reference to `lstat64'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::os::errno':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::cvt':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/mod.rs:(.text._ZN3std3sys4unix2fs4copy17hf7ec4ccda5cbd3e0E+0x53c): undefined reference to `__errno_location'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::fs::copy':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/fs.rs:(.text._ZN3std3sys4unix2fs4copy17hf7ec4ccda5cbd3e0E+0x660): undefined reference to `syscall'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/fs.rs:(.text._ZN3std3sys4unix2fs4copy17hf7ec4ccda5cbd3e0E+0x667): undefined reference to `__errno_location'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::os::errno':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::net::Socket::accept::accept4':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/weak.rs:94: undefined reference to `syscall'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::cvt':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/mod.rs:(.text._ZN3std3sys4unix3net6Socket6accept17hb0113e79d5c7cb35E+0x3f): undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/mod.rs:(.text._ZN3std3sys4unix3net6Socket6accept17hb0113e79d5c7cb35E+0x46): undefined reference to `syscall'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::os::errno':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::pipe::anon_pipe::pipe2':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/weak.rs:94: undefined reference to `syscall'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::os::errno':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::cvt':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/mod.rs:(.text._ZN3std3sys4unix7process13process_inner66_$LT$impl$u20$std..sys..unix..process..process_common..Command$GT$5spawn17h00bfaa1dd31f035cE+0x522): undefined reference to `__errno_location'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::process::process_inner::<impl std::sys::unix::process::process_common::Command>::posix_spawn':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/process/process_unix.rs:336: undefined reference to `posix_spawnattr_init'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/process/process_unix.rs:337: undefined reference to `posix_spawn_file_actions_init'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/process/process_unix.rs:340: undefined reference to `posix_spawn_file_actions_adddup2'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/process/process_unix.rs:345: undefined reference to `posix_spawn_file_actions_adddup2'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/process/process_unix.rs:350: undefined reference to `posix_spawn_file_actions_adddup2'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/process/process_unix.rs:359: undefined reference to `sigemptyset'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/process/process_unix.rs:360: undefined reference to `posix_spawnattr_setsigmask'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/process/process_unix.rs:362: undefined reference to `sigaddset'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/process/process_unix.rs:363: undefined reference to `posix_spawnattr_setsigdefault'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/process/process_unix.rs:368: undefined reference to `posix_spawnattr_setflags'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::process::process_inner::<impl std::sys::unix::process::process_common::Command>::posix_spawn':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1/src/libcore/option.rs:(.text._ZN3std3sys4unix7process13process_inner66_$LT$impl$u20$std..sys..unix..process..process_common..Command$GT$5spawn17h00bfaa1dd31f035cE+0x8b4): undefined reference to `__errno_location'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `<std::sys::unix::process::process_inner::<impl std::sys::unix::process::process_common::Command>::posix_spawn::PosixSpawnattr as core::ops::drop::Drop>::drop':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/process/process_unix.rs:327: undefined reference to `posix_spawnattr_destroy'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `<std::sys::unix::process::process_inner::<impl std::sys::unix::process::process_common::Command>::posix_spawn::PosixSpawnFileActions as core::ops::drop::Drop>::drop':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/process/process_unix.rs:317: undefined reference to `posix_spawn_file_actions_destroy'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::process::process_inner::<impl std::sys::unix::process::process_common::Command>::posix_spawn':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/process/process_unix.rs:374: undefined reference to `posix_spawnp'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `<std::sys::unix::process::process_inner::<impl std::sys::unix::process::process_common::Command>::posix_spawn::PosixSpawnattr as core::ops::drop::Drop>::drop':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/process/process_unix.rs:327: undefined reference to `posix_spawnattr_destroy'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `<std::sys::unix::process::process_inner::<impl std::sys::unix::process::process_common::Command>::posix_spawn::PosixSpawnFileActions as core::ops::drop::Drop>::drop':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/process/process_unix.rs:317: undefined reference to `posix_spawn_file_actions_destroy'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::os::errno':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::cvt':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/mod.rs:(.text._ZN3std3sys4unix7process13process_inner66_$LT$impl$u20$std..sys..unix..process..process_common..Command$GT$7do_exec17heb10be4df6341418E+0x55): undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/mod.rs:(.text._ZN3std3sys4unix7process13process_inner66_$LT$impl$u20$std..sys..unix..process..process_common..Command$GT$7do_exec17heb10be4df6341418E+0x106): undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/mod.rs:(.text._ZN3std3sys4unix7process13process_inner66_$LT$impl$u20$std..sys..unix..process..process_common..Command$GT$7do_exec17heb10be4df6341418E+0x1bb): undefined reference to `__errno_location'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::process::process_inner::<impl std::sys::unix::process::process_common::Command>::do_exec':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/process/process_unix.rs:185: undefined reference to `setgid'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::os::errno':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::process::process_inner::<impl std::sys::unix::process::process_common::Command>::do_exec':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/process/process_unix.rs:197: undefined reference to `setgroups'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/process/process_unix.rs:198: undefined reference to `setuid'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::os::errno':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::process::process_inner::<impl std::sys::unix::process::process_common::Command>::do_exec':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/process/process_unix.rs:217: undefined reference to `sigemptyset'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::os::errno':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `<std::sys::unix::process::process_inner::<impl std::sys::unix::process::process_common::Command>::posix_spawn::PosixSpawnFileActions as core::ops::drop::Drop>::drop':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/process/process_unix.rs:317: undefined reference to `posix_spawn_file_actions_destroy'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `<std::sys::unix::process::process_inner::<impl std::sys::unix::process::process_common::Command>::posix_spawn::PosixSpawnattr as core::ops::drop::Drop>::drop':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/process/process_unix.rs:327: undefined reference to `posix_spawnattr_destroy'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::cvt':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/mod.rs:(.text._ZN3std3sys4unix7process13process_inner7Process4wait17h4329459f1827fe34E+0x49): undefined reference to `__errno_location'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::stack_overflow::imp::drop_handler':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/stack_overflow.rs:178: undefined reference to `sigaltstack'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::stack_overflow::imp::make_handler':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/stack_overflow.rs:156: undefined reference to `sigaltstack'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/stack_overflow.rs:160: undefined reference to `sigaltstack'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::stack_overflow::imp::drop_handler':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/stack_overflow.rs:178: undefined reference to `sigaltstack'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::thread::Thread::set_name':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/thread.rs:96: undefined reference to `prctl'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::thread::guard::current':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/thread.rs:371: undefined reference to `pthread_getattr_np'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/thread.rs:374: undefined reference to `pthread_attr_getguardsize'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/thread.rs:380: undefined reference to `pthread_attr_getstack'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::os::errno':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o):/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: more undefined references to `__errno_location' follow
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::fs::try_statx::statx':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/weak.rs:94: undefined reference to `syscall'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::os::errno':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::fs::try_statx::statx':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/weak.rs:94: undefined reference to `syscall'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(std-73fc3d9b394e10fc.std.6g5uysf2-cgu.0.rcgu.o): In function `std::sys::unix::os::errno':
/rustc/1423bec54cf2db283b614e527cfd602b481485d1//src/libstd/sys/unix/os.rs:62: undefined reference to `__errno_location'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(backtrace-614251cbcc27d069.backtrace.b416pbr5-cgu.0.rcgu.o): In function `backtrace::symbolize::dladdr::resolve':
/cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.40/src/symbolize/dladdr.rs:68: undefined reference to `dladdr'
/cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.40/src/symbolize/dladdr.rs:68: undefined reference to `dladdr'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(alloc.o): In function `__rdos_backtrace_alloc':
alloc.c:(.text.__rdos_backtrace_alloc+0x21): undefined reference to `__errno_location'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(alloc.o): In function `__rdos_backtrace_vector_grow':
alloc.c:(.text.__rdos_backtrace_vector_grow+0x93): undefined reference to `__errno_location'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(alloc.o): In function `__rdos_backtrace_vector_finish':
alloc.c:(.text.__rdos_backtrace_vector_finish+0x39): undefined reference to `__errno_location'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(alloc.o): In function `__rdos_backtrace_vector_release':
alloc.c:(.text.__rdos_backtrace_vector_release+0x32): undefined reference to `__errno_location'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(posix.o): In function `__rdos_backtrace_open':
posix.c:(.text.__rdos_backtrace_open+0x30): undefined reference to `__errno_location'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(posix.o):posix.c:(.text.__rdos_backtrace_close+0x1b): more undefined references to `__errno_location' follow
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(elf.o): In function `__rdos_backtrace_initialize':
elf.c:(.text.__rdos_backtrace_initialize+0xb7): undefined reference to `dl_iterate_phdr'
bazel-out/k8-fastbuild/bin/oak/server/rust/oak_runtime_stripped.a(elf.o): In function `elf_add':
elf.c:(.text.elf_add+0x1186): undefined reference to `__fxstat'
elf.c:(.text.elf_add+0x1205): undefined reference to `__errno_location'
```

</details>

--- a/oak/server/rust/BUILD
+++ b/oak/server/rust/BUILD
@@ -23,9 +23,7 @@ package(
 
 rust_library(
     name = "oak_runtime",
-    srcs = [
-        "src/lib.rs",
-    ],
+    srcs = glob(["src/*.rs"]),
     crate_type = "staticlib",
     edition = "2018",
 )

--- a/oak/server/rust/src/asylo_alloc.rs
+++ b/oak/server/rust/src/asylo_alloc.rs
@@ -1,0 +1,56 @@
+use core::alloc::{GlobalAlloc, Layout};
+
+extern "C" {
+    // Signatures of the functions exposed by the underlying standard library used in Asylo
+    // (newlib). They are statically linked when producing the final enclave executable.
+    //
+    // See https://www.sourceware.org/newlib/libc.html#malloc.
+    fn malloc(nbytes: usize) -> *mut u8;
+    fn memalign(align: usize, nbytes: usize) -> *mut u8;
+    fn free(aptr: *mut u8);
+}
+
+// This aligment is defined in the Intel SGX SDK, and it is used in this allocator to enable a fast
+// path optimization for low alignment values.
+//
+// See the `MALLOC_ALIGNMENT` value for `_TLIBC_` in
+// https://github.com/intel/linux-sgx/blob/d166ff0c808e2f78d37eebf1ab614d944437eea3/sdk/tlibc/stdlib/malloc.c#L541.
+//
+// Also see this paragraph from
+// https://github.com/intel/linux-sgx/blob/d166ff0c808e2f78d37eebf1ab614d944437eea3/sdk/tlibc/stdlib/malloc.c#L248-L253:
+//
+// > MALLOC_ALIGNMENT         default: (size_t)(2 * sizeof(void *))
+// > Controls the minimum alignment for malloc'ed chunks.  It must be a
+// > power of two and at least 8, even on machines for which smaller
+// > alignments would suffice. It may be defined as larger than this
+// > though. Note however that code and data structures are optimized for
+// > the case of 8-byte alignment.
+const MIN_ALIGN: usize = 8;
+
+/// A global memory allocator that links directly to the primitives exposed by Asylo, which in turn
+/// relies on [newlib](http://www.sourceware.org/newlib/libc.html) for standard `malloc` and `free`
+/// calls.
+///
+/// The allocator can be registered with the `#[global_allocator]` attribute (see
+/// https://doc.rust-lang.org/edition-guide/rust-2018/platform-and-target-support/global-allocators.html).
+pub struct System;
+
+// Inspired by https://github.com/apache/incubator-mesatee-sgx/blob/master/sgx_alloc/src/lib.rs.
+unsafe impl GlobalAlloc for System {
+    #[inline]
+    unsafe fn alloc(&self, layout: Layout) -> *mut u8 {
+        // If the alignment requested by the caller is sufficiently small, use a fast path. Since
+        // the alignment is usually a constant at the call site, only the correct branch will be
+        // inlined.
+        if layout.align() <= MIN_ALIGN {
+            malloc(layout.size())
+        } else {
+            memalign(layout.align(), layout.size())
+        }
+    }
+
+    #[inline]
+    unsafe fn dealloc(&self, ptr: *mut u8, _layout: Layout) {
+        free(ptr);
+    }
+}

--- a/oak/server/rust/src/lib.rs
+++ b/oak/server/rust/src/lib.rs
@@ -3,13 +3,43 @@
 
 #![no_std]
 #![feature(lang_items)]
+#![feature(alloc_prelude)]
+#![feature(alloc_error_handler)]
 
+extern crate alloc;
+
+// TODO: Move to separate crate.
+mod asylo_alloc;
+
+use alloc::prelude::v1::*;
+use core::alloc::Layout;
 use core::panic::PanicInfo;
+
+// TODO: Move to separate crate and expose safe wrappers.
+#[link(name = "sgx_trts")]
+extern "C" {
+    // SGX-enabled abort function that causes an undefined instruction (`UD2`) to be executed, which
+    // terminates the enclave execution.
+    //
+    // The C type of this function is `extern "C" void abort(void) __attribute__(__noreturn__);`
+    //
+    // See https://github.com/intel/linux-sgx/blob/d166ff0c808e2f78d37eebf1ab614d944437eea3/sdk/trts/linux/trts_pic.S#L565.
+    fn abort() -> !;
+}
+
+#[global_allocator]
+static A: asylo_alloc::System = asylo_alloc::System;
+
+// Define what happens in an Out Of Memory (OOM) condition.
+#[alloc_error_handler]
+fn alloc_error(_layout: Layout) -> ! {
+    unsafe { abort() }
+}
 
 // See https://doc.rust-lang.org/nomicon/panic-handler.html.
 #[panic_handler]
 fn panic(_info: &PanicInfo) -> ! {
-    loop {}
+    unsafe { abort() }
 }
 
 // See https://doc.rust-lang.org/1.2.0/book/no-stdlib.html.
@@ -20,5 +50,7 @@ pub extern "C" fn eh_personality() {}
 /// It just adds "42" to the provided value and returns it to the caller.
 #[no_mangle]
 pub extern "C" fn add_magic_number(x: i32) -> i32 {
-    x + 42
+    // Allocate a bunch of elements on the heap in order to exercise the allocator.
+    let v: Vec<i32> = (0..10).map(|n| n + 40).collect();
+    x + v[2]
 }

